perception/sub8_perception/src/sub8_perception/start_gate.cc

```diff
@@ -0,0 +1,191 @@
+                    // std::cout << "tl " << gate_line_buffer[i][j][0] << std::endl;
+                    // std::cout << "br " << gate_line_buffer[i][j][1] << std::endl;
+
+                    cv::Point2i s = gate_line_buffer[i][j][1] - gate_line_buffer[i][j][0];
+                    cv::Point2i center = cv::Point2i(s.x / 2 + s.x, s.y / 2 + s.y);
+                    // std::cout << center << std::endl;
+                    accX(center.x);
+                    accY(center.y);
+                    accSizeX(s.x);
+                    accSizeY(s.y);
+                }
+            }
+            // std::cout << mean(acc) << std::endl;
+            // for (int i = 0; i < gate_line_buffer.size(); i++) {
+            //     for (int i = 0; i < gate_line_buffer[0].size(); i++) {
+            // cv::Point2i s = gate_line_buffer[0][i][1] - gate_line_buffer[0][i][0];
+            // cv::Point2i center = cv::Point2i(s.x / 2 + s.x, s.y / 2 + s.y);
+            // std::cout << center << std::endl;
+
+            //         //Make bigger buffer. After like 30 calculate mean and average. Outlier stuff.
+            //     }
+            // }
+            for (size_t i = 0; i < gate_line_buffer[29].size(); i++) {
+                cv::Point2i s = gate_line_buffer[29][i][1] - gate_line_buffer[29][i][0];
+                cv::Point2i center = cv::Point2i(s.x / 2 + s.x, s.y / 2 + s.y);
+                // std::cout << center << std::endl;
+                if (abs(center.x - mean(accX)) > sqrt(variance(accX)) ||
+                        abs(center.y - mean(accY) > sqrt(variance(accY)))) {
+                    gate_line_buffer[29].erase(gate_line_buffer[29].begin() + i);
+                    i--;
+                    std::cout << "Deleted: " << center << std::endl;
+                    continue;
+                }
+                if (abs(s.x - mean(accSizeX)) > sqrt(variance(accSizeX)) ||
+                        abs(s.y - mean(accSizeY)) > sqrt(variance(accSizeY))) {
+                    gate_line_buffer[29].erase(gate_line_buffer[29].begin() + i);
+                    i--;
+                    std::cout << "Deleted size: " << s << std::endl;
+                    continue;
+                }
+            }
+            cv::Mat show = cv::Mat::zeros(rows, cols, CV_8UC3);
+
+            for (size_t i = 0; i < gate_line_buffer[29].size(); i++) {
+                // for (int j = 0; j < gate_line_buffer[0][i].size(); j++) {
+                // cv::line(show, gate_line_buffer[0][i][j], gate_line_buffer[0][i][(j + 1) % 4], cv::Scalar(0, 0, 255), 1, 8 );
+                // }
+                cv::rectangle(show, gate_line_buffer[29][i][0], gate_line_buffer[29][i][1], cv::Scalar(255, 0, 0), 2, 8, 0);
+            }
+            cv::imshow("ay", show);
+            cv::waitKey(30);
+        }
+        findGate(image_msg);
+
+    }
+}
+
+
+void Sub8StartGateDetector::findGate(const sensor_msgs::ImageConstPtr &image_msg) {
```

Comment:
You should comment this so that someone could be able to get the gist of what your algorithm is trying to do when glancing at the code.